src/hooks/useGrantContract.ts

```diff
@@ -0,0 +1,68 @@
+import { ethers } from 'ethers';
+import { useMemo } from 'react';
+import { TokenVestingController__factory } from 'contracts/types';
+
+import { useWeb3React } from '@web3-react/core';
+
+import { GrantContract, Maybe } from '../util';
+/*
+export function useGrantContract(address: string) {
+
+  //Get the contract address
+  let CONTRACT_ADDRESS = address;
+
+  //Connect to Network
+  const provider = new ethers.providers.Web3Provider(window.ethereum);
+
+  //Get Signer
+  const signer = provider.getSigner();
+
+  const grantContract = TokenVestingController__factory.connect(
+    CONTRACT_ADDRESS,
+    signer,
+  );
+
+  return grantContract;
+}*/
+
+export function useGrantContract(address: string): Maybe<GrantContract> {
```

Comment:
This should be integrated inside of the `useContracts` hook. One hook to access all contracts